extractor/src/main/java/org/schabi/newpipe/extractor/stream/StreamInfo.java

```diff
@@ -102,39 +100,46 @@ public static StreamInfo getInfo(final StreamExtractor extractor)
         }
     }
 
-    private static StreamInfo extractImportantData(final StreamExtractor extractor)
+    @Nonnull
+    private static StreamInfo extractImportantData(@Nonnull final StreamExtractor extractor)
             throws ExtractionException {
-        /* ---- important data, without the video can't be displayed goes here: ---- */
-        // if one of these is not available an exception is meant to be thrown directly
-        // into the frontend.
+        // Important data, without it the content can't be displayed.
+        // If one of these is not available, the frontend will receive an exception directly.
 
+        final int serviceId = extractor.getServiceId();
         final String url = extractor.getUrl();
+        final String originalUrl = extractor.getOriginalUrl();
         final StreamType streamType = extractor.getStreamType();
         final String id = extractor.getId();
         final String name = extractor.getName();
         final int ageLimit = extractor.getAgeLimit();
 
-        // suppress always-non-null warning as here we double-check it really is not null
+        // Suppress always-non-null warning as here we double-check it is really not null
```

Comment:
"it really is not null" means the same as "it is really not null" but puts more emphasis on "really being". But anyway ;-)
```suggestion
        // Suppress always-non-null warning as here we double-check it really is not null
```